packages/lit-helpers/src/read-only-properties-mixin.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * @param {any} baseElement
+ */
+export const ReadOnlyPropertiesMixin = baseElement => {
+  /**
+   * @type {Map<string, symbol>}
+   * @private
+   */
+  const _readOnlyPropertyNamesMap = new Map();
+
+  return class extends baseElement {
+    /**
+     * @inheritdoc
+     * @param  {string} name    [description]
+     * @param  {import('lit-element').PropertyDeclaration & { readOnly: Boolean }} options augmented property declaration with optional `readOnly` boolean.
+     */
+    static createProperty(name, options) {
+      let finalOptions = options;
+
+      if (options.readOnly) {
+        const privateName = Symbol(name);
+
+        _readOnlyPropertyNamesMap.set(name, privateName);
+
+        Object.defineProperty(this.prototype, name, {
+          get() {
+            return this[privateName];
+          },
+
+          set(value) {
+            // allow for class field initialization
+            if (this._readOnlyPropertyInitializedMap.get(name)) return;
+            this[privateName] = value;
+            this._readOnlyPropertyInitializedMap.set(name, true);
+          },
+        });
+
+        finalOptions = { ...options, noAccessor: true };
+      }
+
+      super.createProperty(name, finalOptions);
+    }
+
+    /**
+     * @type {Map<string, boolean>}
+     * @private
+     */
+    _readOnlyPropertyInitializedMap = new Map();
+
+    constructor() {
+      super();
+      this.setPropEntry = this.setPropEntry.bind(this);
+    }
+
+    /**
+     * Set read-only properties
+     * @param  {Object<string, unknown>}  props
+     * @return {Promise<this>}
+     * @private
+     */
+    async set(props) {
```

Comment:
`set` is a very generic name, perhaps we can think of something more appropriate to this mixin?

Also is setting multiple properties the common case, or should we make this just key value?

Reply:
maybe `setReadOnlyProperties` would be better.

Reply:
A unary `set` which takes an object mapping is consistent with `React.Component#setState`, which is proof enough that the idiom is popular and well-understood. 

In the project that inspired this, seven out of ten calls to `set` used multiple props. If we didn't provide that here, users would be stuck writing boiler plate again. It's an easy abstraction for use to pick up, with negligible cost against the simple case of a single property updated. By providing an object mapping setter we cater to both case' needs

Reply:
Since you would need to provide the property name anyway that seems reasonable indeed